### PR TITLE
Add types and demo: VictoryBrushContainer

### PR DESCRIPTION
**Pull request by wsparsons**
* Adds a `VictoryBrushContainerDemo` 
* Updates some of the mislabelled props for the demos
* Updates the type definition to use `RangeTuple` instead of `DomainPropType` since it is possible to have the prop to not be provided initially, therefore `DomainPropType` is the wrong type definition due to it requiring either `x` or `y`
* Adds additional missing props for `VictoryBrushContainerProps` interface

### CODE REVIEW

File: packages/victory-zoom-container/src/index.d.ts [hunk 1]
```diff
@@ -10,20 +10,23 @@
...
 
 export interface VictoryZoomContainerProps extends VictoryContainerProps {
   allowPan?: boolean;
   allowZoom?: boolean;
-  clipContainerComponent?: React.ReactElement;
-  zoomDimension?: "x" | "y";
-  zoomDomain?: DomainPropType;
   brushStyle?: React.CSSProperties;
+  clipContainerComponent?: React.ReactElement;
   defaultBrushArea?: "all" | "none" | "disable";
```
Comment (boygirl) [inline packages/victory-zoom-container/src/index.d.ts:20]: `defaultBrushArea` should include the option "move"

File: packages/victory-zoom-container/src/index.d.ts [hunk 2]
```diff
@@ -10,20 +10,23 @@
...
   disable?: boolean;
   downsample?: number | boolean;
   minimumZoom?: CursorData;
```
Comment (boygirl) [inline packages/victory-zoom-container/src/index.d.ts:23]: `CursorData` is a really confusingly named variable. I know you didn't name it, but do you mind changing it?

File: packages/victory-zoom-container/src/index.d.ts (hunk 1 above)
Comment (wsparsons) [reply packages/victory-zoom-container/src/index.d.ts:20]: I currently auditing `Zoom-Container` on a new branch. Would you like me to audit the `Zoom-container` on this branch as well? I figured I break it up in different PR. Also I don't see `defaultBrushArea` as an optional prop in the [docs](https://formidable.com/open-source/victory/docs/victory-zoom-container) nor was it included [here](https://github.com/FormidableLabs/victory/blob/master/packages/victory-zoom-container/src/victory-zoom-container.js#L18-L35).

File: packages/victory-zoom-container/src/index.d.ts (hunk 2 above)
Comment (wsparsons) [reply packages/victory-zoom-container/src/index.d.ts:23]: Sure I can do that

**Comment by wsparsons** [comment]
@boygirl We decided to merge this and keep `VictoryZoomContainer` work separate.

**Pull request merged**